src/components/implementation/cbuf_mgr/default/cbuf_mgr.c

```diff
@@ -755,18 +1009,97 @@ cbuf_register(spdid_t spdid, long cbid)
 
 	printl("cbuf_register\n");
 	CBUF_TAKE();
+	u64_t start, end;
+	rdtscll(start);
+
 	cci = cbuf_comp_info_get(spdid);
-	if (!cci) goto done;
+	if (unlikely(!cci)) goto done;
 	cmr = cbuf_meta_lookup_cmr(cci, cbid);
 	if (cmr) ERR_THROW(cmr->dest, done);
 
 	/* Create the mapping into the client */
-	if (cbuf_alloc_map(spdid, &dest, &p, PAGE_SIZE)) goto done;
-	assert((u32_t)p == round_to_page(p));
+	if (cbuf_alloc_map(spdid, &dest, &p, NULL, PAGE_SIZE, MAPPING_RW)) goto done;
+	assert((unsigned int)p == round_to_page(p));
 	cmr = cbuf_meta_add(cci, cbid, p, dest);
 	assert(cmr);
 	ret = cmr->dest;
 done:
+	rdtscll(end);
+	op_nums[4]++;
+	per_total[4] += (end-start);
+
+	CBUF_RELEASE();
+	return ret;
+}
+
+static void
+cbuf_shrink(struct cbuf_comp_info *cci, int diff)
+{
+	int i, sz;
+	struct cbuf_bin *bin;
+	struct cbuf_info *cbi, *next, *head;
+
+	for(i=cci->nbin-1; i>=0; i--) {
+		bin = &cci->cbufs[i];
+		sz = (int)bin->size;
+		if (!bin->c) continue;
+		cbi = FIRST_LIST(bin->c, next, prev);
+		int i = 0;
+		while (cbi != bin->c) {
```

Comment:
This dos not work. Because we will remove the item from the list inside the cbuf_free_unmap and 
modify the header to the list.